src/Common/src/System/PasteArguments.cs

```diff
@@ -29,18 +30,35 @@ public static string Paste(IEnumerable<string> arguments, bool pasteFirstArgumen
                     //   - Parsing ends at first whitespace outside quoted region.
                     //   - No way to get a literal quote past the parser.
 
+                    bool hasQuote = false;
                     bool hasWhitespace = false;
                     foreach (char c in argument)
                     {
                         if (c == Quote)
                         {
-                            throw new ApplicationException("The argv[0] argument cannot include a double quote.");
+                            hasQuote = true;
                         }
                         if (char.IsWhiteSpace(c))
                         {
                             hasWhitespace = true;
                         }
                     }
+
+                    if (hasQuote)
+                    {
+                        // If the argv[0] contains a double-quote, we're in a pickle because there is no string that would generate such an argv[0] 
+                        // under our parsing rules.
+                        //
+                        // Right now, the only user of this method that asks for arv[0] parsing is Environment.CommandLine. Since it's passing us an arg array that came from
+                        // framework itself as a result of tokenizing a command line under these same rules, this situation should never come up. But if it ever does,
+                        // throwing an exception for this seems extreme given that it's likely that the command line is just being retrieved for logging purposes
```

Comment:
>If the code is reachable, we should have tests for it.

That's a nice principle but 100% code coverage is not attainable in the real world. Some contingency paths are just not practical or cost-effective to recreate in unit tests.

That fact that RemoteInvoker is source-linking this particular one does mean we have a backdoor to test **this** code path (as linked inside S.R.Tests.) It feels a little dodgy but perhaps it's better than no testing at all (and we do lots of "illicit" white-box testing in corefx already.) Let me know what you think about that - I don't want to invest time in that if you're just going to object to it.